### PR TITLE
Feature/msbuild infrastructure improvements

### PR DESCRIPTION
**Pull request by LarsSkovslund**
This PR introduces structural changes to how we define and override global properties and references.

# Structural changes
To unify resolution behavior of `Directory.Build.props` with how `.editorconfig` works, we introduce a section in the existing `Directory.Build.props` that will include upstream `Directory.Build.props` files before applying its own changes. This gives us a central file for common `MSBuild` configurations in the root of a given repository.

> This renders the `Build` folder obsolete.

**Other changes included**
- `Build` folder has been removed and configuration moved to `Directory.Build.props`.
- `directory.build.props` renamed to `Directory.Build.props` to be supported on all platforms by `MSBuild`.
- `directory.build.target` files has been deleted.
- Introduce root level `Directory.Build.props` for solution wide configurations.
- Standard code analyzers inclusion has been moved to root level `Directory.Build.props`.

### CODE REVIEW

**Comment by davidkallesen** [comment]
PR looks fine - but we should wait with approve and merge, until the atc-coding-rules-updater, also have a PR that match this structural change - the have to be in "sync", to avoid issues in projects.
- I will find time in upcomming week, to do the atc-coding-rules-updater improvements.

File: distribution/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <LangVersion>9.0</LangVersion>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>1573,1591,1712</NoWarn>
+
+    <!-- Used by code coverage -->
+    <DebugType>full</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Analyzer settings">
```
Comment (perkops) [inline distribution/Directory.Build.props:24]: Maybe we should add a comment, that this is only relevent for .NET 3.1.x

File: distribution/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Solution wide properties -->
+  <PropertyGroup Label="Assembly Naming">
+    <Company>CompanyName</Company>
+    <Authors>CompanyName</Authors>
+    <Copyright>© $(Company). All rights reserved.</Copyright>
```
Comment (egil) [inline distribution/Directory.Build.props:8]:
```suggestion
```

Probably not necessary in open source projects.

File: distribution/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>1573,1591,1712</NoWarn>
+
+    <!-- Used by code coverage -->
+    <DebugType>full</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Analyzer settings">
+    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
```
Comment (egil) [inline distribution/Directory.Build.props:25]:
```suggestion
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
		<EnableNETAnalyzers>true</EnableNETAnalyzers>
		<AnalysisLevel>latest</AnalysisLevel>
		<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>    
```

Lets turn everything on

**Review by egil — CHANGES_REQUESTED** [review]
Looks good to me. Added a few extra analyzer configurations.

File: distribution/Directory.Build.props (hunk 1 above)
Comment (egil) [reply distribution/Directory.Build.props:23]: are you referring to DebugSymbols?

Comment (perkops) [reply distribution/Directory.Build.props:23]: No - the <AnalysisMode>AllEnabledByDefault</AnalysisMode>

Your checkin changed the line apparently..

Comment (LarsSkovslund) [reply distribution/Directory.Build.props:23]: This is also applicable for net5 [see here](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#analysismode).

Comment (perkops) [reply distribution/Directory.Build.props:23]: > This is also applicable for net5 [see here](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#analysismode).

Cool - think I misread the documentation first time around :)

File: distribution/src/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  <PropertyGroup Label="Documentation">
+    <!--
+    Make sure documentation XML is also included when publishing (not only when testing)
+    see https://github.com/Azure/service-fabric-issues/issues/190
+    -->
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <DocumentationFile Condition=" '$(DocumentationFile)' == '' " >bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+
```
Comment (davidkallesen) [inline distribution/src/Directory.Build.props:33]:
```suggestion
```

Comment (davidkallesen) [reply distribution/src/Directory.Build.props:33]: This should be remove again - GenerateDocumentationFile is already set in root-file

**Pull request merged**